### PR TITLE
Make AzureEP default for python and c# packaging

### PR DESCRIPTION
**Pull request by RandySheriffH**
Make AzureEP default for python and c# packaging, with UT.

### CODE REVIEW

File: tools/ci_build/github/linux/build_linux_arm64_python_package.sh [hunk 1]
```diff
@@ -7,13 +7,16 @@ CXXFLAGS="-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-st
 
 BUILD_DEVICE="CPU"
```
Comment (snnn) [inline tools/ci_build/github/linux/build_linux_arm64_python_package.sh:8]: This script is only used in python packaging pipeline. You can remove the BUILD_DEVICE arg,which was added by you, if we don't need to have two pipelines that one for with AzureEP, one for without AzureEP.

Comment (RandySheriffH) [reply tools/ci_build/github/linux/build_linux_arm64_python_package.sh:8]: No - the device sits there long before AzureEP exists, it is to differentiate hardware distinctions between platforms such as X86 / ARM, hence the name.

File: onnxruntime/test/python/onnxruntime_test_python_azure.py [hunk 2]
```diff
@@ -1,107 +1,14 @@
...
-import numpy as np
-from helper import get_name
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
 
+import unittest
 import onnxruntime as ort
 
 
-class TestAmlEndpoint(unittest.TestCase):
```
Comment (pranavsharma) [inline onnxruntime/test/python/onnxruntime_test_python_azure.py:9]: Why remove so many tests?

Comment (RandySheriffH) [reply onnxruntime/test/python/onnxruntime_test_python_azure.py:9]: The tests are moved to ort extension.
During ort packaging, we do not load extension, meaning this is no way to test inferencing.

File: tools/ci_build/github/linux/build_linux_arm64_python_package.sh (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/linux/build_linux_arm64_python_package.sh:8]: You are right.

**Review by pranavsharma — APPROVED** [review]

**Review by snnn — DISMISSED** [review]

File: onnxruntime/test/python/onnxruntime_test_python_azure.py [hunk 3]
```diff
@@ -1,107 +1,14 @@
...
-        x = np.array([True, False]).astype(bool)
-        y = np.array([True, True]).astype(bool)
-
-        z = sess.run(None, {"X": x, "Y": y}, run_opt)[0]
-
-        expected_z = np.array([True, False]).astype(bool)
-        np.testing.assert_allclose(z, expected_z, rtol=1e-05, atol=1e-08)
+class TestAzureEP(unittest.TestCase):
+    def test_availability(self):
+        self.assertTrue("AzureExecutionProvider" in ort.get_available_providers())
```
Comment (justinchuby) [inline onnxruntime/test/python/onnxruntime_test_python_azure.py:11]:
```suggestion
        self.assertIn("AzureExecutionProvider", ort.get_available_providers())
```

**Review by justinchuby — DISMISSED** [review]

File: onnxruntime/test/python/onnxruntime_test_python_azure.py (hunk 3 above)
Comment (RandySheriffH) [reply onnxruntime/test/python/onnxruntime_test_python_azure.py]: Done.

**Review by snnn — APPROVED** [review]

**Comment by snnn** [comment]
AB#18580

**Pull request merged**